src/server/transaction.cc

```diff
@@ -844,6 +844,14 @@ void Transaction::ExecuteAsync() {
   // IsArmedInShard in other threads.
   run_count_.store(unique_shard_cnt_, memory_order_release);
 
+  EngineShard* shard = EngineShard::tlocal();
```

Comment:
please mention that we can not use `coordinator_index_` because we may offload this function to run in a different thread.

Reply:
nit: using `ServerState::tlocal()->thread_index()` is slightly better because you do not need to check for nullptr.